lint/long-chain_checker.go

```diff
@@ -0,0 +1,119 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/Quasilyte/astcmp"
+)
+
+type longChainChecker struct {
+	ctx *Context
+}
+
+func newLongChainChecker(ctx *Context) Checker {
+	return &longChainChecker{
+		ctx: ctx,
+	}
+}
+
+func (c *longChainChecker) Check(f *ast.File) {
+	for _, decl := range f.Decls {
+		if decl, ok := decl.(*ast.FuncDecl); ok {
+			if decl.Body == nil {
+				continue
+			}
+			for _, stmt := range decl.Body.List {
+				if stmt, ok := stmt.(*ast.SwitchStmt); ok {
+					c.checkSwitch(stmt)
+				}
+			}
+		}
+	}
+}
+
+func (c *longChainChecker) checkSwitch(stmt *ast.SwitchStmt) {
+	exprs := []ast.Expr{}
+	for _, s := range stmt.Body.List {
+		cas := s.(*ast.CaseClause)
+		if cas.List == nil {
+			continue
+		}
```

Comment:
It's actually OK to append nil or empty slice.
Maybe we can remove this if statement completely?
https://play.golang.org/p/tca2ImY64RJ